src/cli/test/task-generate.spec.ts

```diff
@@ -0,0 +1,137 @@
+import type * as d from '../../declarations';
+import { taskGenerate, getBoilerplateByExtension, GenerableExtension } from '../task-generate';
+import { mockConfig, mockStencilSystem } from '@stencil/core/testing';
+import * as utils from '../../utils/validation';
+
+import * as coreCompiler from '@stencil/core/compiler';
+import { CoreCompiler } from '../load-compiler';
+import { Config } from 'prettier';
+
+const promptMock = jest.fn().mockResolvedValue('my-component');
+
+jest.mock('prompts', () => ({
+  prompt: promptMock,
+}));
+
+const setup = async () => {
+  const sys = mockStencilSystem();
+  const config: d.Config = mockConfig(sys);
+  config.configPath = '/testing-path';
+  config.srcDir = '/src';
+
+  // set up some mocks / spies
+  config.sys.exit = jest.fn();
+  config.flags.unknownArgs = [];
+  const errorSpy = jest.spyOn(config.logger, 'error');
+  const validateTagSpy = jest.spyOn(utils, 'validateComponentTag').mockReturnValue(undefined);
+
+  // mock prompt usage: tagName and filesToGenerate are the keys used for
+  // different calls, so we can cheat here and just do a single
+  // mockResolvedValue
+  promptMock.mockResolvedValue({
+    tagName: 'my-component',
+    filesToGenerate: ['css', 'spec.tsx', 'e2e.ts'],
+  });
+
+  return { config, errorSpy, validateTagSpy };
+};
+
+/**
+ * Little test helper function which just temporarily silences
+ * console.log calls so we can avoid spewing a bunch of stuff.
+ */
+async function silentGenerate(coreCompiler: CoreCompiler, config: Config) {
+  const tmp = console.log;
+  const log = jest.fn();
+  console.log = log;
+  await taskGenerate(coreCompiler, config);
+  console.log = tmp;
+  return log;
```

Comment:
Do we need to return this value? It doesn't appear to be used at the moment. If not, I think we can simplify this function by inlining `log`:
```suggestion
  const tmp = console.log;
  console.log = jest.fn();
  await taskGenerate(coreCompiler, config);
  console.log = tmp;
```

Reply:
yeah I had a test which was asserting that the `jest.fn()` was called with some args, but I ended up removing those assertions and forgot to adjust this function